rpcserver.go

```diff
@@ -2661,8 +2661,9 @@ func (state *gbtWorkState) blockTemplateResult(bm *blockManager,
 		}
 
 		// Serialize the transaction for later conversion to hex.
-		txBuf := bytes.NewBuffer(make([]byte, 0, tx.SerializeSize()))
-		if err := tx.Serialize(txBuf); err != nil {
+		var txBuf bytes.Buffer
+		txBuf.Grow(tx.SerializeSize())
```

Comment:
After looking at the trace, it appears that these are no equivalent in terms of allocations, most likely due to the addition of SSA to the compiler.  I actually prefer the single line, but given there is 
no performance hit, I have no objections to the change.